tests/ui/internal-lints/rustc_intrinsic_const_vector_arg_calls.rs

```diff
@@ -0,0 +1,34 @@
+// compile-flags: -Z unstable-options
+#![feature(abi_unadjusted)]
+#![feature(inline_const)]
+#![feature(intrinsics)]
+#![allow(non_camel_case_types)]
+#![feature(repr_simd)]
+#![feature(rustc_attrs)]
+#![feature(simd_ffi)]
+#![allow(unused)]
+
+#[repr(simd)]
+#[derive(Clone)]
+pub struct i8x2(i8, i8);
```

Comment:
You should also test that this works with `repr(simd)` types that wrap `[T; N]` arrays. Please also make sure to test both floats and integers.